### PR TITLE
fix(FEC-10471): adPosition getter returns preroll for mid and post rolls

### PR DESCRIPTION
**Pull request by jesuslc91**
### Description of the Changes

Use `youbora.Constants` instead of `youbora.Adapter.AdPosition`
Also bind to `AD_BREAk_START`

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/adapter/ads/nativeads.js [hunk 1]
```diff
@@ -149,6 +151,7 @@ let NativeAdsAdapter = youbora.Adapter.extend({
 
   loadedAdListener: function (e) {
     this.adObject = e.payload.ad;
+    this.adPosition = e.payload.adType;
```
Comment (yairans) [inline src/adapter/ads/nativeads.js:154]: `adType` is on the ad event just for backward compatibility.
The ad type should be taken from the ad break start event. If it doesn't exist on the ad break start event it's a bug in the player. is this the situation here?

**Review by yairans — CHANGES_REQUESTED** [review]

**Comment by yairans** [comment]
@jesuslc91 Please fill the Description of the Changes

File: src/adapter/ads/nativeads.js (hunk 1 above)
Comment (jesuslc91) [reply src/adapter/ads/nativeads.js:154]: I think hat ad break start event is not even triggered in the environment you provided

Comment (yairans) [reply src/adapter/ads/nativeads.js:154]: adbreakstart triggered. 
What environment you mean?

Comment (jesuslc91) [reply src/adapter/ads/nativeads.js:154]: This one
https://externaltests.dev.kaltura.com/player/library_Player_V3/smartPages/Player_V3_generic_page.php?cdnUrl=test&partnerId=1091&uiConfId=testCanary&selectedUiConf=15225670&entryId=empty&selectedEntryId=0_wifqaipd&compLst=&compVer=&compVers=&logs=deb&pllstByEntId=&pllstId=&cntDwnTTS=&cntDwnDur=&eng1=html&eng2=html&eng3=html&stPr1=&stPr2=&stPr3=&txtLang=default&audioLang=default&a_s=0&strPos=&adTag=noAd&adAftTime=&numRedirects=&dai=noDai&bumper=noBump&bumpPos=defBump&Youbora=on&WMPlace=

Comment (yairans) [reply src/adapter/ads/nativeads.js:154]: this url doesnt contain the ad plugin. use this one
http://externaltests.dev.kaltura.com/player/library_Player_V3/smartPages/Player_V3_generic_page.php?cdnUrl=test&partnerId=1091&uiConfId=testCanary&entryId=empty&chgMed=empty&logs=deb&pllstByEntId=&pllstId=&cntDwnTTS=10&cntDwnDur=5&eng1=html&eng2=html&eng3=html&stPr1=def&stPr2=def&stPr3=def&txtLang=default&audioLang=default&a_s=0&strPos=0&plIn=on&adTag=vmPreMidPost&adAftTime=&Youbora=on&WMPlace=botRight

Comment (jesuslc91) [reply src/adapter/ads/nativeads.js:154]: True, sorry, that wasn't the link with ads. 
For the one you sent I don't see that event being triggered neither.

Comment (yairans) [reply src/adapter/ads/nativeads.js:154]: for sure it triggered. but as i see in the your code `startBreakAdListener` doesn't bound to the event.
bind it to `AD_BFREAk_START` in `registerListeners`

Comment (jesuslc91) [reply src/adapter/ads/nativeads.js]: Updating

File: src/adapter/ads/nativeads.js [hunk 2]
```diff
@@ -28,24 +28,26 @@ let NativeAdsAdapter = youbora.Adapter.extend({
...
-        returnValue = youbora.Adapter.AdPosition.POSTROLL;
-        break;
-      case 'midroll':
-        break;
-      case 'overlay':
-        returnValue = 'overlay';
-        break;
-      default:
-        if (!this.plugin.getAdapter().flags.isJoined) {
+    if (this.adPosition) {
```
Comment (yairans) [inline src/adapter/ads/nativeads.js:31]: I think this if-else logic is equal to the switch-default case so you can revert it.

**Review by yairans — CHANGES_REQUESTED** [review]

File: src/adapter/ads/nativeads.js [hunk 3]
```diff
@@ -27,25 +27,27 @@ let NativeAdsAdapter = youbora.Adapter.extend({
...
+        case 'midroll':
+          break;
+        case 'overlay':
+          returnValue = 'overlay';
+          break;
+      }
+    } else {
+      if (!this.plugin.getAdapter().flags.isJoined) {
+        returnValue = youbora.Constants.AdPosition.Preroll;
+      } else if (!this.plugin.getIsLive() && this.plugin.getPlayhead() > this.plugin.getDuration() - 1) {
```
Comment (yairans) [inline src/adapter/ads/nativeads.js:48]: Why did you remove `getAdapter()` here? is it related to this issue?

File: src/adapter/ads/nativeads.js [hunk 4]
```diff
@@ -118,7 +120,8 @@ let NativeAdsAdapter = youbora.Adapter.extend({
       [Event.AD_THIRD_QUARTILE]: this.thirdQuartileListener.bind(this),
       [Event.ENTER_FULLSCREEN]: this.enterFullscreenListener.bind(this),
       [Event.EXIT_FULLSCREEN]: this.exitFullscreenListener.bind(this),
-      [Event.AD_MANIFEST_LOADED]: this.manifestLoaded.bind(this)
+      [Event.AD_MANIFEST_LOADED]: this.manifestLoaded.bind(this),
+      [Event.AD_BREAK_START]: this.startBreakAdListener.bind(this)
```
Comment (yairans) [inline src/adapter/ads/nativeads.js:124]: seems like `endBreakAdListener` also not bound to its event - `AD_BREAK_END`

**Comment by OrenMe** [comment]
@jesuslc91 the branch is out-of-date, please align. Thanks.

File: src/adapter/ads/nativeads.js (hunk 3 above)
Comment (jesuslc91) [reply src/adapter/ads/nativeads.js]: No is not, now the object plugin is accessing to the adapter methods instead (safely, checking that both the plugin and the method exist) so we can save some code while adding an extra check

**Review by yairans — APPROVED** [review]
